### DISCUSSION TITLE
Control MQTT topic

### DISCUSSION THREAD

**Discussion by cliftonarms** [fl4p/batmon-ha#176] [Q&A]
How can I set a root topic of "BMS" and name each one 1,2,3....n

I have multiple BMS and instead of having name based MQTT topics of 
    BMS1, BMS2, BMS3....BMSn, 

is it possible to have mqtt topics of 
    BMS/1/, BMS/2/, BMS/3/, BMS/n/

Any advice on recommended practice to pre-append mqtt topic in batmon-ha would be great

**Comment by cliftonarms** [comment]
Anyone - Please ?

**Reply by fl4p** [reply]
can you elaborate why you need that?
why doesn't BMS_1, BMS_2 .. prefix doesn't work?

**Reply by cliftonarms** [reply]
I have 12 BMS's. I would like them under the same MQTT parent topic of "BMS" and each individual bms device on topic "BMS/1/"... "BMS/2/"... "BMS/2/"........ "BMS/12/ etc

This will enable MQTT wildcard "BMS/+/sub_topic"  to be used to test and subscribe to all BMS specific sub topics and perform efficient monitoring

i.e. subscribing to "BMS/+/Current" to monitor instantaneous current across all bms and perform checks

Hope this clarifies - and thankyou for your assistance.

**Reply by fl4p** [reply]
Understood. I've pushed a change that allows slashes in the mqtt topic prefix https://github.com/fl4p/batmon-ha/commit/2a7851382de1365861a0f731a48ea888ff45ce44 

Try and let me know wether it works

**Reply by cliftonarms** [reply]
@fl4p Hi - can you commit this update to the HA build please.

Then I can test the modification you have made

**Please**

**Reply by cliftonarms** [reply]
This change correctly creates the MQTT topics with hierarchical slashes "/" 

But it does not create the Home assistant sensors correctly, in fact no sensors are created at all.

**Reply by Karatzie** [reply]
Hi, I would to create the first 9 BMSn with 01, 02, 03 ... 09 and then 10, 11, 12 and so on.
This should be better for the sorting of the BMS number.

I have a Bluetti with 16 serial cells.
I used this numbering for the voltage of the cells.

**Comment by chuxxsss** [comment]
Hi,
   I have added Batmon to my HA intergation.  Now I go into my Utility meter help and I have no mqtt sensors listed. Now saying that I was using a dongle to monitor 433 through mqtt would that cause a problem?  By the way thank you for making this software.  All I need after this is it to talk to my glowatt though wifi then.

Kind regards 
Chris
